### PR TITLE
LF: remove versionning of the individual values in transaction proto.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
We now use directly the version from the node.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thank you!

File: daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto [hunk 1]
```diff
@@ -49,17 +49,25 @@ message Node {
 }
 
 message KeyWithMaintainers {
-    com.daml.lf.value.VersionedValue key = 1;
+    oneof key {
+        com.daml.lf.value.VersionedValue key_versioned = 1;
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto]: We indeed care only of the binary compatibility. 
1 - I think it is better to gave explanatory name 
2 - I do not provide any form of backward compatibility for whoever wants to use this protobuf to write his own reader.

[the remark this answers is not in the document]
Comment (cocreature) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto]: perfectly reasonable here. I think for ledger API protobufs we would probably have to handle this differently but for this, this is fine.

File: daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto [hunk 2]
```diff
@@ -49,17 +49,25 @@ message Node {
...
+    }
     repeated string maintainers = 2; // the maintainers induced by the key
 }
 
 message NodeCreate {
     reserved 1; // was contract_id
-    ContractInstance contract_instance = 2;
+    oneof contract {
+        ContractInstance contract_instance = 2;  // defined in version < dev
+        com.daml.lf.value.Value arg_unversioned = 7; // defined in version >= dev
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:63]: I do not like contract instance. 
1. I think contract Instance is a somehow a wrong concept. Why `agreementText` is a more import than the signatories or the key. 
2. I plan to flatten the contract instance inside the Scala `NodeCreate`. 
3. by flattening the contract instance, we get the invariant that each node has a field templateId
4. we save a bit of space like that.

[the remark this answers is not in the document]
Comment (cocreature) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:63]: Good argument so let’s go with your proposal.

[the remark this answers is not in the document]
Comment (cocreature) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:63]: Actually if we go with your proposal do we need the `oneof` at all? Can we just add the argument as a separate field given that all fields are optional in protobuf? I think the part that I find confusing here is that we have a oneof and then a bunch of flattened extra fields and they are related.

[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto:63]: Fine that.

**Review by stefanobaghino-da — COMMENTED** [review]
Thanks for taking care of the benchmarks as well. It's probably worth running the benchmarks to see if there is any effect at this point.

**Pull request merged**